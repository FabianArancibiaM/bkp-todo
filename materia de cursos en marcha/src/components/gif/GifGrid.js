import React from 'react'
import { useFetchGifs } from '../../hooks/useFetchGifs';
import {GifGridItem} from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);
    
    return (
        <div className="card-grid animate__animated animate__fadeIn">
            <h3>{category}</h3><br/>
                {loading && <p className="card-grid animate__animated animate__flash" >Loading...</p> }
                {
                    images.map( (img)=> 
                        <GifGridItem 
                        key={img.id}
                        { ...img } 
                        />
                    )
                }
        </div>
    )
}

