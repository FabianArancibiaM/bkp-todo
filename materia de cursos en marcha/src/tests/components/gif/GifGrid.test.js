import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGrid } from '../../../components/gif/GifGrid';
describe('GifGrid Test Funciones', () => {
    let wraper = shallow(<GifGrid />);
    beforeEach( () => {
        wraper = shallow(<GifGrid />);
    });
    test('GifGrid', () => {
        expect(wraper).toMatchSnapshot();
    })
})