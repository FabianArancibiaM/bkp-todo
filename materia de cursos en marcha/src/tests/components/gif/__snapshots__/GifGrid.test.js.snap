// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Test Funciones GifGrid 1`] = `
<div
  className="card-grid animate__animated animate__fadeIn"
>
  <h3 />
  <br />
  <p
    className="card-grid animate__animated animate__flash"
  >
    Loading...
  </p>
</div>
`;
